### PR TITLE
update filter.py

### PR DESCRIPTION
**Pull request by Nov00786**
Changing the "sep=\n" to "sep=\0" so the error (ValueError: Specified \n as separator or delimiter. This forces the python engine which does not accept a line terminator. Hence it is not allowed to use the line terminator as separator.) won't appear.

### CODE REVIEW

File: filtering/filter.py [hunk 1]
```diff
@@ -117,9 +117,9 @@ def prepare(source_file, target_file, source_lang, target_lang, lower=False):
     df_source = df["Source"]
     df_target = df["Target"]
 
-    df_source.to_csv(source_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\n")
+    df_source.to_csv(source_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\0")
     print("--- Source Saved:", source_file)
-    df_target.to_csv(target_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\n")
+    df_target.to_csv(target_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\0")
```
Comment (AaronCGoidel) [inline filtering/filter.py]: This change causes `TypeError: "delimiter" must be a 1-character string`. It seems to me that it should remain `"\n"` when writing.

Comment (ymoslem) [reply filtering/filter.py]: I agree with @AaronCGoidel that "\0" can be used on in reading, but not for writing. Could you please update it @Nov00786 

For consistency, I think we should do the same for the two files under [train_dev_split](https://github.com/ymoslem/MT-Preparation/tree/main/train_dev_split)

Many thanks!

Comment (Nov00786) [reply filtering/filter.py]: updated

Comment (ymoslem) [reply filtering/filter.py]: @Nov00786  We agreed to keep it as "\0" in `pd.read_csv`, right?

**Comment by ymoslem** [comment]
Okay, I edited it to be "\0" in `read_csv` but "\n" in `to_csv`. If there is any other issue, please let me know. Thanks!

File: filtering/filter.py (hunk 1 above)
Comment (Nov00786) [reply filtering/filter.py]: @ymoslem I apologize you are right it should be \0 in reading. thank you for correcting my mistake

Comment (ymoslem) [reply filtering/filter.py]: Thanks to you @Nov00786 and @AaronCGoidel!

**Pull request closed without merging**